be/src/formats/parquet/column_reader.cpp

```diff
@@ -257,6 +263,14 @@ class ScalarColumnReader : public ColumnReader {
     const TypeDescriptor* _col_type = nullptr;
     const tparquet::ColumnChunk* _chunk_metadata = nullptr;
     std::unique_ptr<ColumnOffsetIndexCtx> _offset_index_ctx;
+
+    // _can_lazy_decode means string type and all page dict code
```

Comment:
It seems this optimization does not only works for string type but decimal, double types, etc too

Reply:
yes, but we have not support support reading dict_code for decimal and double types, and we also don't support bytes->decimal convert for parquet, so up to now, we don't need to do this.